src/core/stream/period/period_stream.ts

```diff
@@ -157,36 +165,46 @@ export default function PeriodStream({
       const newStream$ = clock$.pipe(
         take(1),
         mergeMap((tick) => {
+          const { audioTrackSwitchingMode } = options;
           const qSourceBuffer = createOrReuseQueuedSourceBuffer(sourceBuffersStore,
                                                                 bufferType,
                                                                 adaptation,
                                                                 options);
           const strategy = getAdaptationSwitchStrategy(qSourceBuffer,
                                                        period,
                                                        adaptation,
-                                                       tick);
+                                                       tick,
+                                                       audioTrackSwitchingMode);
           if (strategy.type === "needs-reload") {
             return observableOf(EVENTS.needsMediaSourceReload(period, tick));
           }
 
           const cleanBuffer$ = strategy.type === "clean-buffer" ?
             observableConcat(...strategy.value.map(({ start, end }) =>
                                qSourceBuffer.removeBuffer(start, end))
-                            ).pipe(ignoreElements()) :
-            EMPTY;
+                            ).pipe(last())
```

Comment:
I needed `last()` operator, to know when the observable has finished emitting and so to complete because that observable `cleanBuffer$` doesn't emit at all `Observable<never>`, I also needed to emit an event once finished.

I don't really see how I can do it with the `concat()` operator?

Reply:
To know when the observable has completed.

Like instead of:
```js
cleanBuffer$.pipe(last(), mergeMap(() => doSomethingAfter$));
```
It can just be:
```js
concat(cleanBuffer$, doSomethingAfter$);
```
As long as we don't need what cleanBuffer$ is emitting, which is the case here

Reply:
Oh right! as the `concat` operator emit in order! Right! thanks!